### PR TITLE
punnett -> Latextablemaker

### PR DESCRIPTION
**Pull request by remyslab**
Hi!
Check my version of your punnettsquaremaker, I added a feature to produce Latex tables.

Cheers, 
Rémy, from France.

### CODE REVIEW

**Comment by remyslab** [comment]
I added a python3 version :)

File: README.md [hunk 1]
```diff
@@ -1,9 +1,25 @@
-PunnetSquareMaker
-=================
+Punnett Square Maker
```
Comment (catherinemoresco) [inline README.md]: Would you mind leaving the spaces out of the title?

File: punnetsquaremaker.py [hunk 2]
```diff
@@ -15,7 +15,28 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 # The author can be reached by email at catherine.moresco@gmail.com.
+# Improved with Latex tables by Remyslab, email at remyslab@laposte.net.
 
+import datetime
+
+def headtex():
+	f= open(name, 'w')
```
Comment (catherinemoresco) [inline punnetsquaremaker.py]: nit: please use spaces on either side of equals signs (here and in the rest of the files)

https://google.github.io/styleguide/pyguide.html?showone=Whitespace#Whitespace

File: punnetsquaremaker.py [hunk 3]
```diff
@@ -39,40 +60,77 @@ def make_table(parent1, parent2):
...
 while True:
+	now = datetime.datetime.now() # Date & time
+	name = '%s_table.tex'%(now.strftime("%Y-%m-%d_%H-%M-%S")) # Give a name for the file 
 	p1 = raw_input("Please enter the genotype of the first parent: ").split(' ')
 	p2 = raw_input("Please enter the gentype of the second parent: ").split(' ')
+	headtex()
 	c1 = get_all_combinations(p1)
 	c2 = get_all_combinations(p2)
+	
+	f= open(name, 'a')
```
Comment (catherinemoresco) [inline punnetsquaremaker.py]: Please consider making the writing of a LaTex file optional--i.e., giving the user the ability to use the tool from the command line without creating a new file.

File: punnetsquaremaker.py [hunk 4]
```diff
@@ -15,7 +15,28 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 # The author can be reached by email at catherine.moresco@gmail.com.
+# Improved with Latex tables by Remyslab, email at remyslab@laposte.net.
 
+import datetime
+
+def headtex():
```
Comment (catherinemoresco) [inline punnetsquaremaker.py:22]: Please consider refactoring so the file is only opened/closed once--opening and closing files is a relatively expensive operation. For example, you might want to make these functions return strings, and then write them to the file all at once in the main loop.

https://stackoverflow.com/questions/11349020/python-file-open-close-everytime-vs-keeping-it-open-until-the-process-is-finishe

File: punnetsquaremaker.py [hunk 5]
```diff
@@ -39,40 +60,77 @@ def make_table(parent1, parent2):
...
 	a = make_table(c1, c2)
 	print_table(a, c1, c2)
+	foottex()
+	freqhead()
 	print_genotype_frequencies(a)
-	action = raw_input("Enter (Q) to quit, or (A) to make another!\n")
-	if action == "Q":
+	foottex()
+	print '' 
+	print 'Your Latex file ' + name + ' is saved in your script repertory !\n'
```
Comment (catherinemoresco) [inline punnetsquaremaker.py]: nit: remove spaces between words and exclamation points, here and below.

**Review by catherinemoresco — CHANGES_REQUESTED** [review]
Thanks so much for contributing! LaTex is the best, it's really awesome to have support for that.

File: README.md (hunk 1 above)
Comment (remyslab) [reply README.md]: Ok for "PunnetSquareMaker".

File: punnetsquaremaker.py (hunk 4 above)
Comment (remyslab) [reply punnetsquaremaker.py:22]: Ok, I'll fix it asap.

File: punnetsquaremaker.py (hunk 2 above)
Comment (remyslab) [reply punnetsquaremaker.py]: Ok, I'll fix it asap.

File: punnetsquaremaker.py (hunk 3 above)
Comment (remyslab) [reply punnetsquaremaker.py]: Ok, I'll fix it asap.

File: punnetsquaremaker.py (hunk 5 above)
Comment (remyslab) [reply punnetsquaremaker.py]: Ok, I'll fix it asap.

**Comment by remyslab** [comment]
Thank you Catherine, this's my first contribution with a Github repo (and also I've just few skills in coding...). Your review is appreciated, it makes me better for the next time :)
I'll fix your request as soon as possible and and try to make a new pull request.
Rémy

**Comment by remyslab** [comment]
Done, I hope there aren't mistakes?

**Review by catherinemoresco — APPROVED** [review]

**Comment by catherinemoresco** [comment]
Thanks Remy! Looks good. Congrats on your first pr! :)

**Pull request merged**